yt_dlp/extractor/nebula.py

```diff
@@ -245,30 +286,40 @@ class NebulaChannelIE(NebulaBaseIE):
                 'description': 'Enjoy these hottest of takes on Disney, Transformers, and Musicals.',
             },
             'playlist_mincount': 2,
+        }, {
+            'url': 'https://nebula.tv/johnnyharris',
+            'info_dict': {
+                'id': 'johnnyharris',
+                'title': 'Johnny Harris',
+                'description': 'I make videos about maps and many other things.',
+            },
+            'playlist_mincount': 90,
         },
     ]
 
-    def _generate_playlist_entries(self, collection_id, channel):
-        episodes = channel['episodes']['results']
-        for page_num in itertools.count(2):
-            for episode in episodes:
-                yield self._build_video_info(episode)
-            next_url = channel['episodes']['next']
-            if not next_url:
-                break
-            channel = self._call_nebula_api(next_url, collection_id, auth_type='bearer',
-                                            note=f'Retrieving channel page {page_num}')
-            episodes = channel['episodes']['results']
-
     def _real_extract(self, url):
         collection_id = self._match_id(url)
-        channel_url = f'https://content.watchnebula.com/video/channels/{collection_id}/'
-        channel = self._call_nebula_api(channel_url, collection_id, auth_type='bearer', note='Retrieving channel')
-        channel_details = channel['details']
+        first_api_response = self._call_nebula_api(
+            f'https://content.watchnebula.com/video/channels/{collection_id}/', collection_id,
+            auth_type='bearer', note='Retrieving channel')
+
+        page_api_response = first_api_response
+
+        def page_func(page_num):
+            nonlocal page_api_response
+
+            if page_num > 0:
+                next_url = page_api_response['episodes'].get('next')
+                if not next_url:
+                    return []
+
+                page_api_response = self._call_nebula_api(
+                    next_url, collection_id, auth_type='bearer', note=f'Downloading page {page_num}')
+
+            return [self.url_result(episode['share_url'], ie=NebulaIE, **self._extract_video_metadata(episode))
+                    for episode in page_api_response['episodes']['results']]
```

Comment:
This is completely wrong. The point of `PagedList` is that the pages can be requested out of order (e.g. when you pass `-I 100-110`). You cannot assume `page_api_response` is the response from previous page. Since the pagination happens using `next` key, this seems unsuitable for a PagedList and should just be a generator like original code

Reply:
>  The point of PagedList is that the pages can be requested out of order 
It would be great to add it to the class documentation; it 's not obvious, at least for me

https://github.com/yt-dlp/yt-dlp/blob/master/yt_dlp/utils.py#L2917-L2921

Reply:
Any specific suggestion?

Reply:
I imagine InAdvancePagedList is similar, so better in the base class. Something like this:

```python
class PagedList:
    """
        Entries generator for Offset-based pagination APIs.
        e.g. https://api.example.com/v2/videos?offset=200&limit=50

        NOTE: This should not be used with Token-based pagination APIs,
        because it's essential being able to retrieve a specific page
        without having to request any previous one.
    """
```